schema/anyof.go

```diff
@@ -25,8 +25,7 @@ func (v AnyOf) Compile(rc ReferenceChecker) error {
 // Validate ensures that at least one sub-validator validates.
 func (v AnyOf) Validate(value interface{}) (interface{}, error) {
 	for _, validator := range v {
-		var err error
-		if value, err = validator.Validate(value); err == nil {
+		if value, err := validator.Validate(value); err == nil {
```

Comment:
The initial problem here is that we where overwriting the `value` functional parameter, so that the next iteration of the for-loop would use the value returned by the previous (failing) validator, which in most cases would be nil due to the returned error.

With this fix, we generate a _new_ (shadowing) value variable for each iteration. This variable is not passed on to the next iteration.